src/chain.ts

```diff
@@ -163,7 +167,13 @@ export const chain = (
             head: clientHead,
         });
 
-        sourceSocket.write(isPlain ? '' : `HTTP/1.1 200 Connection Established\r\n\r\n`);
+        try {
+            sourceSocket.write(isPlain ? '' : `HTTP/1.1 200 Connection Established\r\n\r\n`);
+        } catch (error) {
+            targetSocket.destroy();
+            sourceSocket.destroy(error as Error);
+            return;
+        }
```

Comment:
Please comment on why is try catch needed

Reply:
Added.